### PR TITLE
gdb.debug() environment variables fix

### PR DESCRIPTION
**Pull request by jlewis200**
If passed an empty dictionary defining the environment variables, the process invoked by gdb.debug() has environment variable 'PWD' set to the directory containing the executable.  This behaviour is not observered when NOPTRACE is enabled:  the program runs without any environment variables set.  Similarly, if an empty environment variable dictionary is passed to process(), it appears to also run under a truly empty environment.  Additionally, attempts to overwrite the "PWD" environment variable with a dictionary entry are not successful under gdb.debug().

This fix brings the gdb.debug() treatment of environment variables in line with process() and gdb.debug() with NOPTRACE enabled. 
 It allows both empty environments, and overwriting of the 'PWD' environment variable.

# Pwntools Pull Request

Thanks for contributing to Pwntools!  Take a moment to look at [`CONTRIBUTING.md`][contributing] to make sure you're familiar with Pwntools development.

Please provide a high-level explanation of what this pull request is for.

## Testing

Pull Requests that introduce new code should try to add doctests for that code.  See [`TESTING.md`][testing] for more information.

## Target Branch

Depending on what the PR is for, it needs to target a different branch.

You can always [change the branch][change] after you create the PR if it's against the wrong branch.

| Branch   | Type of PR                                                       |
| -------- | ---------------------------------------------------------------- |
| `dev`    | New features, and enhancements
| `dev`    | Documentation fixes and new tests
| `stable` | Bug fixes that affect the current `stable` branch
| `beta`   | Bug fixes that affect the current `beta` branch, but not `stable`
| `dev`    | Bug fixes for code that has never been released

[contributing]: https://github.com/Gallopsled/pwntools/blob/dev/CONTRIBUTING.md
[testing]: https://github.com/Gallopsled/pwntools/blob/dev/TESTING.md
[change]: https://github.com/blog/2224-change-the-base-branch-of-a-pull-request

### CODE REVIEW

File: pwnlib/gdb.py [hunk 1]
```diff
@@ -236,13 +236,14 @@ def _gdbserver_args(pid=None, path=None, args=None, which=None, env=None):
     if pid:
         gdbserver_args += ['--once', '--attach']
 
-    if env:
+    if env != None:
```
Comment (Arusekk) [inline pwnlib/gdb.py]: Identity comparison is faster and generally recommended by Python docs when dealing with `None`:
```suggestion
    if env is not None:
```

**Review by Arusekk — CHANGES_REQUESTED** [review]
Great! I did not know that there is a difference. Please add my change, and I will merge this.

**Comment by jlewis200** [comment]
> Great! I did not know that there is a difference. Please add my change, and I will merge this.

Ok, I commited the identity comparison changes you suggested.  It appears to be a gdbserver quirk.  gdbserver is launched in the specified environment, but the debugged process always always has PWD set unless using the env wrapper with the -i option..

**Comment by heapcrash** [comment]
I merged this since it's a useful way to set the environment, but I can't reproduce a single-env-var process with `gdb.debug`.

```py
ELF.from_assembly(shellcraft.sh()).debug(env={'foo': 'bar\x01\x02\x03\xff'})
```

Still shows PWD

```
pwndbg> env
00:0000│   0xffc7581c —▸ 0xffc75fad ◂— 0x3d6f6f66 ('foo=')
01:0004│   0xffc75820 —▸ 0xffc75fb9 ◂— 0x3d445750 ('PWD=')
02:0008│   0xffc75824 ◂— 0x0
```

**Comment by jlewis200** [comment]
> I merged this since it's a useful way to set the environment, but I can't reproduce a single-env-var process with `gdb.debug`.
> 
> ```python
> ELF.from_assembly(shellcraft.sh()).debug(env={'foo': 'bar\x01\x02\x03\xff'})
> ```
> 
> Still shows PWD
> 
> ```
> pwndbg> env
> 00:0000│   0xffc7581c —▸ 0xffc75fad ◂— 0x3d6f6f66 ('foo=')
> 01:0004│   0xffc75820 —▸ 0xffc75fb9 ◂— 0x3d445750 ('PWD=')
> 02:0008│   0xffc75824 ◂— 0x0
> ```

I see what you mean.  This method seems to work fine when used with preexisting executables, but not the ELF.from_assembly() case you showed above.

**Pull request merged**